src/components/AnalysisName.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Button, Input } from 'semantic-ui-react'
+import * as app from '../store/actions/app'
+
+class AnalysisName extends React.Component {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isEditing: false
+    }
+
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleClick = this.handleClick.bind(this)
+    this.handleCancel = this.handleCancel.bind(this)
+  }
+
+  handleSubmit (event) {
+    let input = this.refs.viewName.inputRef.value
+    if (input === '') {
+      input = 'Untitled Analysis'
```

Comment:
This is also a little more friendly to future stuff -- if we ever need to know, for instance, that the name is empty, we can check for an empty string instead of having to check for the string "Untitled Analysis." This is especially helpful if (a) we ever change the string, or (b) the application is ever localized and now you have the equivalent of "Untitled Analysis" in multiple languages.